wandb/sdk/wandb_run.py

```diff
@@ -2160,6 +2160,9 @@ def _finish(
         quiet: Optional[bool] = None,
     ) -> None:
         logger.info(f"finishing run {self._get_path()}")
+        if self.disabled:
+            return
```

Comment:
why not just update the definition in `_make_run_disabled`? i feel like we need to clean this code path a lot, now this disable is scattered all over and the source of our issues :(